packages/p2p/src/components/modal-manager/modals/my-ads-floating-rate-switch-modal/my-ads-floating-rate-switch-modal.tsx

```diff
@@ -1,25 +1,22 @@
 import React from 'react';
-import { Localize } from 'Components/i18next';
 import { Button, Modal } from '@deriv/components';
-import { observer } from 'mobx-react-lite';
+import { observer } from '@deriv/stores';
+import { Localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 import { ad_type } from 'Constants/floating-rate';
 import { useStores } from 'Stores';
-import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 
 const MyAdsFloatingRateSwitchModal = () => {
     const { floating_rate_store, my_ads_store } = useStores();
-    const { is_modal_open } = useModalManagerContext();
+    const { hideModal, is_modal_open } = useModalManagerContext();
+    const { rate_type, reached_target_date } = floating_rate_store;
+    const { selected_ad_type } = my_ads_store;
```

Comment:
```suggestion
    const { selected_ad_type, toggleMyAdsRateSwitchModal } = my_ads_store;
```

Reply:
i did try destructuring this.. but it didn't work.. there was this.something inside the togglemyadsrateswitchmodal.. and this got undefined..

Reply:
you need to add action.bound to toggleMyAdsRateSwitchModal for `this` to work

Reply:
yes. it has been updated.